backend/src/validation/workspace.ts

```diff
@@ -57,8 +58,11 @@ export const validateClientForWorkspace = async ({
         environment,
         requiredPermissions
       });
-
-      return {};
+      return { membership, workspace};
+    case ActorType.SERVICE_V3:
```

Comment:
how come we default to UnauthorizedRequestError for this when for v2 service token we try to validate it

Reply:
Because this function is deprecated.

We are stripping out all middlewares because we moved it to controller-level validation so I figured to just put this in place for now since I will be deleting this function from the codebase entirely.

Reply:
It’s also not used anywhere anymore and where ST V3 might be relevant.